grpc/server/src/backend.cpp

```diff
@@ -0,0 +1,104 @@
+#include "backend.h"
+
+#include <grpc/grpc.h>
+#include <grpc++/server.h>
+#include <grpc++/server_builder.h>
+#include <grpc++/server_context.h>
+#include <grpc++/security/server_credentials.h>
+
+#include "action/actionrpc_impl.h"
+#include "core/corerpc_impl.h"
+#include "log.h"
+#include "mission/missionrpc_impl.h"
+#include "telemetry/telemetryrpc_impl.h"
+
+namespace dronecore::backend {
+
+int DroneCoreBackend::run(const int mavlink_listen_port)
+{
+    if (int returnCode = connect_to_vehicle(mavlink_listen_port) != 0) {
+        return returnCode;
+    }
+
+    grpc::ServerBuilder builder;
+    setup_port(builder);
+
+    CoreServiceImpl core(dc);
+    builder.RegisterService(&core);
+
+    Action action(&dc.device());
+    ActionServiceImpl actionService(action);
+    builder.RegisterService(&actionService);
+
+    Mission mission(&dc.device());
+    MissionServiceImpl missionService(mission);
+    builder.RegisterService(&missionService);
+
+    Telemetry telemetry(&dc.device());
+    TelemetryServiceImpl telemetryService(telemetry);
+    builder.RegisterService(&telemetryService);
+
+    server = builder.BuildAndStart();
+    LogInfo() << "Server started";
+    server->Wait();
+
+    return 0;
+}
+
+int DroneCoreBackend::connect_to_vehicle(const int port)
+{
+    if (!add_udp_connection(port)) {
+        return 1;
+    }
+
+    log_uuid_on_timeout();
+    wait_for_discovery();
```

Comment:
return value omitted.

Reply:
Good catch

Reply:
Right, I forgot to log the UUID on discovery. Thanks!

Reply:
Actually I'm not sure I understand this comment... what do you mean?

Reply:
Good catch :-).

Reply:
Ok! I said that you have omitted returning an `int` value from this function.